lib/puppet_x/puppetlabs/scheduled_task/taskscheduler2.rb

```diff
@@ -71,6 +71,12 @@ module TaskScheduler2
   TASK_FLAG_SYSTEM_REQUIRED              = 0x1000
   TASK_FLAG_RUN_ONLY_IF_LOGGED_ON        = 0x2000
 
+  def self.is_com_error(win32OLERuntimeError, hresult)
+    # does not include 0x prefix
+    hex = (hresult & Error::MAX_32_BIT_VALUE).to_s(16)
```

Comment:
Seems a little silly to take the original hex value, convert to 2s complement for storage... and then revert to `0x800XXXXX` style for a string comparison.

Maybe better to just store as `0x800XXXXX` and add a helper to `Error` class to use that format when numbers need to be compared.